pages/selfCheckQuestions.tsx

```diff
@@ -0,0 +1,272 @@
+                <Text ml={10} mr={2} fontWeight="bold" fontSize="4xl">
+                    Create a Tool
+                </Text>
+                <Flex wrap={"wrap"} ml={"auto"} mr={10}>
+                    <Button
+                        _hover={{ bg: "#F3F3F3" }}
+                        _active={{
+                            transform: "scale(0.95)",
+                        }}
+                        onClick={onOpen}
+                        minWidth={"90"}
+                        colorScheme="white"
+                        variant="outline"
+                    >
+                        Discard
+                    </Button>
+                    <Button
+                        _hover={{ bg: "#121310" }}
+                        _active={{
+                            transform: "scale(0.95)",
+                        }}
+                        ml={"5"}
+                        minWidth={"90"}
+                        color="white"
+                        background="black"
+                        variant="outline"
+                        //TODO: Send this output to the database
+                        //rather than just logging it in the console
+                        onClick={() => {
+                            alert("Check the console for the object");
+                            console.log(questionList);
+                        }}
+                    >
+                        Save
+                    </Button>
+                </Flex>
+            </Flex>
+            <Modal
+                blockScrollOnMount={false}
+                isOpen={isOpen}
+                onClose={onClose}
+                motionPreset="slideInBottom"
+            >
+                <ModalOverlay />
+                <ModalContent p={"5"}>
+                    <ModalHeader>Delete Tool </ModalHeader>
+                    <ModalCloseButton />
+                    <ModalBody>
+                        Are you sure you want to discard this tool? This is a
+                        permanent action that cannot be undone.
+                    </ModalBody>
+                    <ModalFooter>
+                        <Button
+                            variant="outline"
+                            colorScheme="black"
+                            mr={3}
+                            w={100}
+                            onClick={onClose}
+                        >
+                            Cancel
+                        </Button>
+                        <Button
+                            onClick={() => {
+                                onClose();
+                                removeAllQuestions();
+                            }}
+                            w={100}
+                            colorScheme="red"
+                        >
+                            Discard
+                        </Button>
+                    </ModalFooter>
+                </ModalContent>
+            </Modal>
```

Comment:
If you want to check out @daniel-sm-yu's work on the global Modal component (#27) , it might be useful to modify this into using that. I know it does essentially the same work so I'll leave this as your decision